plugins/node/instrumentation-fs/test/parent.test.ts

```diff
@@ -0,0 +1,164 @@
+    assert.strictEqual(memoryExporter.getFinishedSpans().length, 0);
+  };
+
+  beforeEach(() => {
+    const contextManager = new AsyncHooksContextManager();
+    api.context.setGlobalContextManager(contextManager.enable());
+  });
+
+  afterEach(() => {
+    plugin.disable();
+    memoryExporter.reset();
+  });
+
+  const createSpanTests = (shouldCreateSpan: boolean) => {
+    const prefix = shouldCreateSpan ? 'creates' : 'does not create';
+    const expectedSpans = () => (shouldCreateSpan ? 1 : 0) + additionalSpans;
+
+    it(`${prefix} a span with the callback API`, async () => {
+      await new Promise<void>(resolve => {
+        api.context.with(context, () => {
+          fs.access('./test/fixtures/readtest', fs.constants.R_OK, () =>
+            resolve()
+          );
+        });
+      }).then(() => endRootSpan());
+
+      assert.deepEqual(
+        memoryExporter.getFinishedSpans().length,
+        expectedSpans()
+      );
+    });
+
+    it(`${prefix} a span with the synchronous API`, () => {
+      api.context.with(context, () => {
+        fs.accessSync('./test/fixtures/readtest', fs.constants.R_OK);
+        endRootSpan();
+      });
+
+      assert.deepEqual(
+        memoryExporter.getFinishedSpans().length,
+        expectedSpans()
+      );
+    });
+
+    if (supportsPromises) {
+      it(`${prefix} a span with the promises API`, async () => {
+        await new Promise<void>(resolve => {
+          api.context.with(context, () => {
+            fs.promises
+              .access('./test/fixtures/readtest', fs.constants.R_OK)
+              .finally(() => resolve(endRootSpan()));
+          });
+        });
+
+        assert.deepEqual(
+          memoryExporter.getFinishedSpans().length,
+          expectedSpans()
+        );
+      });
+    }
+  };
+
+  const withRootSpan = (fn: () => void) => {
+    describe('with a root span', () => {
+      beforeEach(() => {
+        const rootSpan = tracer.startSpan('root span');
+
+        context = api.trace.setSpan(api.context.active(), rootSpan);
+        endRootSpan = () => rootSpan.end();
+        additionalSpans = 1;
+      });
+
+      fn();
+    });
+  };
+
+  const withoutRootSpan = (fn: () => void) => {
+    describe('without a root span', () => {
+      beforeEach(() => {
+        context = api.context.active();
+        endRootSpan = () => {};
+        additionalSpans = 0;
+      });
+
+      fn();
+    });
+  };
+
+  describe('requireParentSpan enabled', () => {
+    beforeEach(() => {
+      initializePlugin({ requireParentSpan: true });
+    });
+
+    withRootSpan(() => {
+      createSpanTests(true);
+    });
+
+    withoutRootSpan(() => {
+      createSpanTests(false);
+    });
+  });
+
+  describe('requireParentSpan disabled', () => {
+    beforeEach(() => {
+      initializePlugin({ requireParentSpan: false });
+    });
+
+    withRootSpan(() => {
+      createSpanTests(true);
+    });
+
+    withoutRootSpan(() => {
+      createSpanTests(true);
```

Comment:
optional: When reading this, one has to jump to the function definition to understand the meaning of `true` and `false` for the `createSpanTests` invocation. A possible alternative is to use `createSpanTests({expectFsSpan: true})` which can be considered more readable, similar to how `initializePlugin({ requireParentSpan: false });` communicates the intention directly